openeo/rest/datacube.py

```diff
@@ -1141,6 +1141,40 @@ def reduce_dimension(
             metadata=self.metadata.reduce_dimension(dimension_name=dimension) if self.metadata else None,
         )
 
+    @openeo_process
+    def reduce_spatial(
+        self,
+        reducer: Union[str, typing.Callable, UDF, PGNode],
+        context: Optional[dict] = None,
+        process_id="reduce_spatial",
+        band_math_mode: bool = False,
+    ) -> "DataCube":
+        """
+        Add a reduce process with given reducer callback along the spatial dimensions
+
+        :param reducer: the "child callback":
+            the name of a single openEO process,
+            or a callback function as discussed in :ref:`callbackfunctions`,
+            or a :py:class:`UDF <openeo.rest._datacube.UDF>` instance.
+
+            The callback should correspond to a process that
+            receives an array of numerical values
+            and returns a single numerical value.
+            For example:
+
+            -   ``"mean"`` (string)
+            -   :py:func:`absolute <openeo.processes.max>` (:ref:`predefined openEO process function <openeo_processes_functions>`)
+            -   ``lambda data: data.min()`` (function or lambda)
+
+        :param context: Additional data to be passed to the process.
+        """
+        reducer = self._get_callback(process=reducer, parent_parameters=["data", "context"], connection=self.connection)
+
+        return self.process_with_node(
```

Comment:
I don't think it's necessary to use `process_with_node` here. Just using `self.process()` will do.

Reply:
I just tried. Using self.process leads to an error. `process_with_node` work fine instead.

```
File ~\Documents\GitHub\openeo-python-client\openeo\internal\graph_building.py:298, in FlatGraphNodeIdGenerator.generate(self, process_id)
    296 def generate(self, process_id: str):
    297     """Generate new key for given process id."""
--> 298     self._counters[process_id] += 1
    299     return "{p}{c}".format(p=process_id.replace('_', ''), c=self._counters[process_id])

TypeError: unhashable type: 'PGNode'
```

Reply:
I guess you tried something like `self.process(PGNode(...`
This should work however:
```python
self.process(
    process_id="reduce_spatial",
    data=THIS,
    reducer=reducer,
    context=context,
    metadata=...
```